python/example_code/neptune/neptune_scenario.py

```diff
@@ -0,0 +1,748 @@
+
+
+# snippet-end:[neptune.python.describe.cluster.main]
+
+# snippet-start:[neptune.python.describe.dbinstance.main]
+def check_instance_status(neptune_client, instance_id: str, desired_status: str):
+    """
+    Polls the status of a Neptune DB instance until it reaches desired_status.
+    Uses pagination via describe_db_instances — even for a single instance.
+
+    Raises:
+      ClientError: If describe_db_instances fails (e.g., instance not found).
+      RuntimeError: If timeout expires before reaching desired status.
+    """
+    paginator = neptune_client.get_paginator('describe_db_instances')
+    start_time = time.time()
+
+    while True:
+        try:
+            # Paginate responses for the specified instance ID
+            pages = paginator.paginate(DBInstanceIdentifier=instance_id)
+            instances = []
+            for page in pages:
+                instances.extend(page.get('DBInstances', []))
+        except ClientError:
+            # Let the calling code handle errors such as ResourceNotFound
+            raise
+
+        current_status = instances[0].get('DBInstanceStatus') if instances else None
+        elapsed = int(time.time() - start_time)
+
+        print(f"\rElapsed: {format_elapsed_time(elapsed)}  Status: {current_status}", end="", flush=True)
+
+        if current_status and current_status.lower() == desired_status.lower():
+            print(f"\nInstance '{instance_id}' reached '{desired_status}' in {format_elapsed_time(elapsed)}.")
+            return
+
+        if elapsed > TIMEOUT_SECONDS:
+            raise RuntimeError(f"Timeout waiting for '{instance_id}' to reach '{desired_status}'")
+
+        time.sleep(POLL_INTERVAL_SECONDS)
+
+
+# snippet-end:[neptune.python.describe.dbinstance.main]
+
+# snippet-start:[neptune.python.create.dbinstance.main]
+def create_db_instance(neptune_client, db_instance_id: str, db_cluster_id: str) -> str:
+    try:
+        request = {
+            'DBInstanceIdentifier': db_instance_id,
+            'DBInstanceClass': 'db.r5.large',
+            'Engine': 'neptune',
+            'DBClusterIdentifier': db_cluster_id
+        }
+
+        print(f"Creating Neptune DB Instance: {db_instance_id}")
+        response = neptune_client.create_db_instance(**request)
+
+        instance = response.get('DBInstance')
+        if not instance or 'DBInstanceIdentifier' not in instance:
+            raise RuntimeError("Instance creation succeeded but no ID returned.")
+
+        # Wait for it to become available
+        print(f"Waiting for DB Instance '{db_instance_id}' to become available...")
+        waiter = neptune_client.get_waiter('db_instance_available')
+        waiter.wait(
+            DBInstanceIdentifier=db_instance_id,
+            WaiterConfig={'Delay': 30, 'MaxAttempts': 40}
+        )
+
+        print(f"DB Instance '{db_instance_id}' is now available.")
+        return instance['DBInstanceIdentifier']
+
+    except ClientError as e:
+        raise ClientError(
```

Comment:
This should handle a ClientError, not raise one. This code will swallow all other errors, regardless of their type, which is incorrect.